fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -52,8 +54,25 @@ class SignalSparsity:
             conv.weight.data = 3d_sparser.get_sst_dst_weight(conv.weight.data)
     """
 
-    def __init__(self) -> None:
-        pass
+    def __init__(
+        self,
+        algo: Algo = Algo.FFT,
+        sst_top_k_dim: Optional[int] = -1,
+        sst_top_k_element: Optional[int] = None,
+        sst_top_k_percent: Optional[float] = 0.1,
+        dst_top_k_dim: Optional[int] = -1,
+        dst_top_k_element: Optional[int] = None,
+        dst_top_k_percent: Optional[float] = 0.1,
```

Comment:
@min-xu-ai For both the `sst_top_k_percent` and `dst_top_k_percent`, does it seem that starting with the default as `None` might be more appropriate? Since, we only want one of `top_k_element` and `top_k_percent` to have a value, having a default value forces the caller to proactively choose `dst/sst_top_k_percent = None.` 

E.g. `sparser_2d = SignalSparsity(sst_top_k_element=100, sst_top_k_percent=None, sst_top_k_dim=1, .....)` 
Having all `topk_k_percent` and `topk_k_element` arguments as `None` alleviates this problem.

Reply:
I thought the default is (element=None, percent=0.1), which is a common case since user may want to common do a percent based topk? Are you saying you want to make the common case with percent=None and element=x? What's the x value by default?

Reply:
No really, I am saying (element=None, percent=None) seems like a better choice to me. 

1. For example, with (element=None, percent=0.1) default, say if we want to instantiate with (element = 10), we actually need to instantiate it with (element=10, percent=None), because we cannot allow (element = 10, percent=0.1) implicitly.

Reply:
I see. That's true. Ignore my other comment on the default value here then. I was thinking default to user not supplying any of the 2 args. But your idea here is potentially better. It is just that user must know one of the arg is "required". Default case will error out. Make sure the docstring is really clear for them and error msg direct them to how to fix the error when they use both None as the defaults.